hail/Makefile

```diff
@@ -19,9 +19,14 @@ HAIL_PATCH_VERSION := 136
 HAIL_PIP_VERSION := $(HAIL_MAJOR_MINOR_VERSION).$(HAIL_PATCH_VERSION)
 HAIL_VERSION := $(HAIL_PIP_VERSION)-$(SHORT_REVISION)
 
-SCALA_VERSION ?= 2.12.18
+SCALA_VERSION ?= 2.12
 SPARK_VERSION ?= 3.5.0
 
+ALLOWED_SCALA_VERSIONS := 2.12.13 2.12 2.13
```

Comment:
Why `2.12.13`?

Reply:
https://github.com/hail-is/hail/blob/main/build.yaml#L937

Reply:
For reference, the mill build now hard codes a few supported scala versions, because they're encoded by a cross-module now instead of having the entire build depend on a scala version environment variable. That lets us have some source files that are only for specific scala versions, and to have separate subdirectories in `out/` for different versions so we don't have to recompile from scratch when switching.

I made `2.12` and `2.13` lookup a minor version in `build.mill`, so bumping a minor version can happen in one place. `2.12.13` is the only specific minor version we were building with, for Azure.